Objects/complexobject.c

```diff
@@ -948,9 +940,7 @@ complex_new_impl(PyTypeObject *type, PyObject *r, PyObject *i)
                      "complex() first argument must be a string or a number, "
                      "not '%.200s'",
                      Py_TYPE(r)->tp_name);
-        if (own_r) {
-            Py_DECREF(r);
-        }
+        assert(!own_r);
```

Comment:
Please add a comment explaining why this assertion is currently valid. (The validity depends on significant context, some of which is external to this module, and some of that context may change in the future.)